plugins/font/plugin.js

```diff
@@ -1,45 +1,75 @@
-﻿/**
+/**
  * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
  * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
  */
 
 ( function() {
-	function addCombo( editor, comboName, styleType, lang, entries, defaultLabel, styleDefinition, order ) {
+	function addCombo( editor, comboName, styleType, lang, entries, defaultLabel, configStyleDefinition, order ) {
 		var config = editor.config,
-			style = new CKEDITOR.style( styleDefinition );
+			allowedAndRequiredContent = new CKEDITOR.style( configStyleDefinition ),
+			commandName = comboName.slice( 0, 1 ).toLowerCase() + comboName.slice( 1 ),
+			preparedStylesAndNames = _prepareStylesAndNames( {
+				entries: entries,
+				styleType: styleType,
+				configStyleDefinition: configStyleDefinition
+			} ),
+			styles = preparedStylesAndNames.styles,
+			names = preparedStylesAndNames.names;
+
+		editor.addCommand( commandName , {
+			contextSensitive: true,
+			exec: function( editor, data ) {
+				if ( editor.readOnly ) {
+					return;
+				}
 
-		// Gets the list of fonts from the settings.
-		var names = entries.split( ';' ),
-			values = [];
+				var newStyle = data.newStyle,
+					oldStyle = data.oldStyle,
+					range = editor.getSelection().getRanges()[ 0 ],
+					isRemove = newStyle === undefined;
 
-		// Create style objects for all fonts.
-		var styles = {};
-		for ( var i = 0; i < names.length; i++ ) {
-			var parts = names[ i ];
+				// Prevent of throw error when situation from this bug appear (#1116).
+				if ( !oldStyle && !newStyle ) {
+					return;
+				}
 
-			if ( parts ) {
-				parts = parts.split( '/' );
+				// If the range is collapsed we can't simply use the editor.removeStyle method
+				// because it will remove the entire element and we want to split it instead.
+				if ( oldStyle && range.collapsed ) {
+					_splitElementOnCollapsedRange( {
+						editor: editor,
+						range: range,
+						style: oldStyle
+					} );
+				}
 
-				var vars = {},
-					name = names[ i ] = parts[ 0 ];
+				if ( isRemove ) {
+					editor.removeStyle( oldStyle );
+				} else {
+					if ( oldStyle ) {
+						editor.removeStyle( oldStyle );
+					}
 
-				vars[ styleType ] = values[ i ] = parts[ 1 ] || name;
+					editor.applyStyle( newStyle );
+				}
+			},
 
-				styles[ name ] = new CKEDITOR.style( styleDefinition, vars );
-				styles[ name ]._.definition.name = name;
-			} else {
-				names.splice( i--, 1 );
+			refresh: function( editor, path ) {
+				if ( !allowedAndRequiredContent.checkApplicable( path, editor, editor.activeFilter ) ) {
+					this.setState( CKEDITOR.TRISTATE_DISABLED );
+				} else {
+					this.setState( CKEDITOR.TRISTATE_OFF );
+				}
 			}
-		}
+		} );
 
 		editor.ui.addRichCombo( comboName, {
 			label: lang.label,
 			title: lang.panelTitle,
 			toolbar: 'styles,' + order,
-			defaultValue: 'cke-default',
```

Comment:
I'm wondering why default value has been set into `cke-default` in the first place and if it's safe to replace it with an empty string. It looks like later it's added to richcombo entries as an index key, so it will result in object like:

```javascript
items = {
'': '(Default)',
'Arial': 'Arial'
...
}
```
It's a valid object key because it's just expected that an object key will be just a string type, however, it may be error-prone because the empty string is also a falsy value. Imagine something like:

```javascript
for ( var key in items ) {
  // This will never execute for empty string key,
  // however, from the code logic point of view, it's correct key with some value.
 if ( key ) {
    console.log( items[ key ] );
}
```

This may work for now, but later it could create hard to find issue. Also, if it's done on purpose, so the default value is skipped for some algorithms, it still seems better to explicitly check if it's the default value for better readability. Could you tell the reason for this change?

Reply:
It was suggested here: https://github.com/ckeditor/ckeditor4/pull/3375#pullrequestreview-283067208
However, I haven't noticed this scenario.

Reply:
Oh, shoot, sorry for that. Self-destruction:smile: But that's what happens when continuing review a couple months later :cry: